### PR TITLE
[IAM] [Doc] Update Deployment Guide and Using WSO2 Identity Server sections

### PR DESCRIPTION
**Pull request by YvonneW**
**Renamed and updated the following files:** 

- deployment-patterns.md
- fronting-with-the-nginx-load-balancer.md
- setting-up-separate-databases-for-clustering.md
- configuring-the-datasources.md
- deployment-guidelines-in-production.md
- using-apis.md
- calling-admin-services.md
- admin-services-for-one-way-operations.md
- rest-apis.md
- using-the-account-recovery-rest-apis.md
- using-the-self-sign-up-rest-apis.md
- using-the-authentication-rest-apis.md
- using-the-consent-management-rest-apis.md
- using-the-personal-information-export-rest-apis.md
- using-the-configuration-management-rest-apis.md
- retrieving-tenant-resources-based-on-search-parameters.md
- using-the-scim-2.0-rest-apis.md
- using-the-openid-connect-dynamic-client-registration-rest-apis.md
- using-oauth2-scopes-rest-apis.md
- authenticating-and-authorizing-rest-apis.md
- authentication-data-api.md
- soap-apis.md
- user-management-with-apis.md
- managing-users-and-roles-with-apis.md
- managing-claims-with-apis.md
- managing-permissions-with-apis.md
- counting-users-and-roles-using-management-console-and-admin-services.md
- managing-user-stores-with-apis.md
- user-information-recovery-service.md
- identity-application-management-api.md
- using-the-service-provider-api.md
- service-provider-configurations-used-with-apis.md
- identity-provider-management-api.md
- using-the-identity-provider-api.md
- identity-provider-configurations-used-with-apis.md
- scim-1.1-apis.md
- managing-tenants-with-apis.md
- developer-guide.md
- working-with-the-source-code.md
- extending-the-identity-server.md
- user-management-for-developers.md
- writing-a-custom-password-validator.md
- writing-a-custom-claim-handler.md
- user-store-listeners.md
- using-the-user-management-errors-event-listener.md
- carbon-remote-user-store-manager.md 
- working-with-errors.md
- customizing-authentication-error-messages.md
- customizing-error-pages.md
- error-codes-and-descriptions.md
- provisioning-for-developers.md
- writing-an-outbound-provisioning-connector.md
- extensible-scim-user-schemas-with-wso2-identity-server.md
- identity-federation-for-developers.md
- writing-a-custom-federated-authenticator.md
- writing-a-custom-local-authenticator.md
- access-control-for-developers.md
- writing-a-custom-policy-info-point.md
- workflow-management-for-developers.md
- writing-a-custom-event-handler.md
- writing-a-custom-workflow-template.md
- adaptive-authentication-for-developers.md
- writing-custom-functions-for-adaptive-authentication.md

**Renamed the following images and moved them to appropriate folders:**

Deployment Guide

- attachments/103329476/103329481.png ==> ../../assets/img/setup/deployment-with-the-loadbalancer.png
- attachments/103329499/103329500.png ==>  ../../assets/img/setup/data-types.png
- attachments/56984556/61672631.png ==> ../../assets/img/setup/artifact-management.png
- attachments/103329463/103329465.png ==>  ../../assets/img/setup/scenario.png
- attachments/103329463/103329464.png ==> ../../assets/img/setup/kestores.png attachments
- attachments/103329466/103329467.png ==> ../../assets/img/setup/synchronous-token-persistence.png
- attachments/103329466/103329468.png ==> ../../assets/img/setup/asynchronous-token-persistence.png

Calling Admin Services

../../assets/img/103330060/103330061.png ==> ../../assets/img/using-wso2-identity-server/admin-services-list.png

Using the Configuration Management REST APIs
../../assets/img/119111748/119111749.png ==> ../../assets/img/using-wso2-identity-server/configuration-management-architecture.png

Counting users and roles using Management Console and Admin Services

- ../../assets/img/103330053/103330054.png ==> ../../assets/img/using-wso2-identity-server/users.png
- ../../assets/img/103330053/103330055.png ==> ../../assets/img/using-wso2-identity-server/roles.png
- 
- Customizing Login Pages for Service Providers
- ../../assets/img/103330112/103330125.png ==> ../../assets/img/using-wso2-identity-server/travelocity-screen.png  
- ../../assets/img/103330112/103330124.png ==> ../../assets/img/using-wso2-identity-server/avis-screen.png
- ../../assets/img/103330112/112392626.png ==> ../../assets/img/using-wso2-identity-server/add-new-service-provider.png
- ../../assets/img/103330112/112392631.png ==> ../../assets/img/using-wso2-identity-server/saml2-web-sso-configuration-option.png
- ../../assets/img/103330112/112392636.png ==> ../../assets/img/using-wso2-identity-server/new-application-screen.png 
- ../../assets/img/103330112/103330122.png ==> ../../assets/img/using-wso2-identity-server/identity-server-sign-in-screen.png
- ../../assets/img/103330112/103330120.png ==> ../../assets/img/using-wso2-identity-server/web-app-folder-strcture.png
- ../../assets/img/103330112/103330130.png ==> ../../assets/img/using-wso2-identity-server/travelocity-login-screen.png

Customizing the Authentication Endpoint

- ../../assets/img/103330101/103330108.png ==> ../../assets/img/using-wso2-identity-server/registry-browse-menu-item.png
- ../../assets/img/103330101/103330107.png){height="250"} ==> ../../assets/img/using-wso2-identity-server/registry-browser.png
- ../../assets/img/103330101/103330106.png ==> ../../assets/img/using-wso2-identity-server/add-resource-option.png
- ../../assets/img/103330101/103330105.png ==> ../../assets/img/using-wso2-identity-server/add-resource-form.png
- ../../assets/img/103330101/103330104.png ==> ../../assets/img/using-wso2-identity-server/registry-resource.png
- ../../assets/img/103330101/103330103.png ==> ../../assets/img/using-wso2-identity-server/resource-properties.png
- ../../assets/img/103330101/103330102.png ==> ../../assets/img/using-wso2-identity-server/add-new-property-option.png

User Store Listeners

- /assets/img/103330086/103330088.png ==> /assets/img/using-wso2-identity-server/user-store-operations.png
- Carbon Remote User Store Manager
- ../../assets/img/103330074/103330075.png ==> ../../assets/img/using-wso2-identity-server/carbon-remote-user-store-manager.png
- ../../assets/img/103330074/103330077.png ==>  ../../assets/img/using-wso2-identity-server/add-new-user-store.png
- Extensible SCIM User Schemas With WSO2 Identity Server
- ../../assets/img/29923055/29906299.png ==> ../../assets/img/using-wso2-identity-server/sample-user-object.png
- ../../assets/img/29923055/29906300.png ==> ../../assets/img/using-wso2-identity-server/sample-user-object.png
- ../../assets/img/29923055/29906301.png ==> ../../assets/img/using-wso2-identity-server/sample-extended-user-object.png
- ../../assets/img/29923055/29906302.png ==> ../../assets/img/using-wso2-identity-server/new-user-attributes.png
- ../../assets/img/29923055/29906303.png ==> ../../assets/img/using-wso2-identity-server/created-scim-user-object.png
- Writing a Custom Local Authenticator
- ../../assets/img/103330069/103330070.png ==> ../../assets/img/using-wso2-identity-server/oauth-openid-connect-configuration-option.png
- Writing a Custom Event Handler
- ../../assets/img/103330095/103330097.png ==> ../../assets/img/using-wso2-identity-server/handler-hierarchy.png
- ../../assets/img/103330095/103330098.png ==> ../../assets/img/using-wso2-identity-server/operation-category-selection.png
- ../../assets/img/103330095/103330096.png ==> ../../assets/img/using-wso2-identity-server/operation-category-name.png

**Added the following images:**

- ../../assets/img/callbacl-url.jpeg
- ../../assets/img/using-wso2-identity-server/local-authentication-check-box.jpeg
- ../../assets/img/using-wso2-identity-server/playground-app.jpeg

**Renamed the following attachments and moved them to appropriate folders:** 

Admin Services for One Way Operations

- ../../assets/img/112364676/112364673.wsdl ==> ../../assets/attachments/user-identity-management-admin-service.wsdl
- ../../assets/img/112364676/112364672.wsdl ==> ../../assets/attachments/remote-user-store-manager-service.wsdl
- ../../assets/img/112364676/112364674.wsdl ==> ../../assets/attachments/user-information-recovery-service.wsdl
- ../../assets/img/112364676/112364675.wsdl ==> ../../assets/attachments/authentication-admin-service.wsdl
- ../../assets/img/112364676/112364671.wsdl ==> ../../assets/attachments/challenge-question-management-admin-service.wsdl

Customizing the SSO Redirection Page
../../assets/img/103330131/103330132.html ==> ../../assets/attachments/sample-sso-redirect-page-template.html

### CODE REVIEW

File: en/mkdocs.yml [hunk 1]
```diff
@@ -78,19 +78,17 @@ nav:
...
+      - 'Deployment Patterns': 
         - 'Deployment Patterns': setup/deployment-patterns.md
-        - 'Setting Up Deployment Pattern 1': setup/setting-up-deployment-pattern-1.md
-        - 'Setting Up Deployment Pattern 2': setup/setting-Up-Deployment-Pattern-2.md
-        - 'Setting Up Separate Databases for Clustering': setup/setting-Up-Separate-Databases-for-Clustering.md
-      - 'Deployment Guidelines in Production': setup/deployment-Guidelines-in-Production.md
+        - 'Fronting with the Nginx load balancer': setup/fronting-with-the-nginx-load-balancer.md
+        - 'Configuring the Datasource' :setup/configuring-the-datasources.md
+        - 'Setting Up Separate Databases for Clustering': setup/setting-up-separate-databases-for-clustering.md
+      - 'Deployment Guidelines in Production': setup/deployment-suidelines-in-Production.md
```
Comment (sherenewso2) [inline en/mkdocs.yml]: typo - suidelines should be changed to guidelines

File: en/mkdocs.yml [hunk 2]
```diff
@@ -342,33 +340,35 @@ nav:
...
           - 'Password Recovery':
-            - 'Password Recovery': using-wso2-identity-server/password-recovery.md
-            - 'Configuring Google reCaptcha for Security-Question Based Password Recovery': using-wso2-identity-server/configuring-google-recaptcha-for-security-question-based-password-recovery.md
-            - 'Configuring reCaptcha for Password Recovery': using-wso2-identity-server/configuring-recaptcha-for-password-recovery.md
-          - 'Managing Challenge Questions': using-wso2-identity-server/managing-challenge-questions.md
+            - 'Password Recovery': using-wso2-identity-server/password-Recovery.md
+            - 'Configuring Google reCaptcha for Security-Question Based Password Recovery': using-wso2-identity-server/configuring-Google-reCaptcha-for-Security-Question-Based-Password-Recovery.md
+            - 'Configuring reCaptcha for Password Recovery': using-wso2-identity-server/configuring-reCaptcha-for-Password-Recovery.md
+          - 'Managing Challenge Questions': using-wso2-identity-server/managing-Challenge-Questions.md
```
Comment (sherenewso2) [inline en/mkdocs.yml]: the .md file names need to be reverted back to simple case. please fix in all the relevant places

File: en/docs/setup/adding-high-availability-for-ldap.md [hunk 3]
```diff
@@ -12,18 +12,21 @@ Server when you add high availability to your LDAP.
...
 
-    !!! note
-    
+2.  Failover for LDAP is configured by defining LDAP connection URLs with a space between them in the `<IS_HOME>/repository/conf/deployment.toml` file.
+
+    !!! example 
+        ```
+        [user_store]
+        ...
+        connection_url = "ldap://localhost:10500 ldap://localhost:10511"
```
[the remark this answers is not in the document]
Comment (YvonneW) [reply en/docs/setup/adding-high-availability-for-ldap.md]: To show that there could be content before that.

File: en/docs/setup/configuring-the-datasources.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    !!! tip
+        Alternatively, you can create more databases for each type of data to separate the data logically. Note that this will NOT make a difference in performance and is not actually neccessary. 
+        However, if you do wish to separate the data logically into separate databases, see the [Setting Up Separate Databases for Clustering](../../setup/setting-up-separate-databases-for-clustering) topic.
+
+2. Configure the datasource for the databases in both nodes of your cluster in the  `master-datasources.xml`   file found in the  `<IS_HOME>/repository/conf/datasources` folder.  The code block below shows a sample configuration of the user mangement database and identity database for a mysql database.  For instructions on how to configure the datasource depending on the type of database you created, see [Changing the Carbon Database](../../admin-guide/changing-the-carbon-database).
+
+    ??? example "Click to see the sample configuration of master-datasource.xml"
+        ```xml
```
[the remark this answers is not in the document]
Comment (YvonneW) [reply en/docs/setup/configuring-the-datasources.md]: Yet to be defined.

**Comment by YvonneW** [comment]
Closing this PR due to too many conflicts.

**Pull request closed without merging**